src/pages/chatRoom.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useState, useEffect, useRef } from 'react';
+import NavBar from '../components/navBar';
+import DateDivider from '../components/dateDivider';
+import MyMessageBox from '../components/myMessageBox';
+import OtherMessageBox from '../components/otherMessageBox';
+import PlusButton from '../components/plusButton';
+import ChatHeader from '../components/chatHeader';
+import messagesData from '../data/chatData.json';
+import ProfileButton from '../components/profileButton';
+
+// Message 타입 정의 : 타입 스크립트 쓰니까 편하네요!!!
+type Message = {
+    messageId: number;
+    // 메세지 아이디를 따로 저장해두면 좋을 것 같아서 일단 넣어줬습니다! 습관 비스무리한..
+    senderId: number;
+    // 사용자 전환 시 메세지 디자인도 변경도 변경됩니다. 이를 위해 senderId를 넣어줬습니다!
+    message: string;
+    // 메세지 내용은 string으로 저장했습니다!
+    timestamp: string;
+    // 일단 시간을 데이터로 저장하기 위해 string으로 저장했습니다!
+};
+
+const ChatRoom: React.FC = () => {
+    // 현재 사용자 ID 상태 (기본값: 1)
+    const [userId, setUserId] = useState<number>(1);
+
+    // 메시지 데이터 상태 (초기값: 빈 배열)
+    const [messages, setMessages] = useState<Message[]>([]);
+
+    // 컴포넌트 첫 랜더링시 JSON 데이터를 상태에 설정함
+    useEffect(() => {
+        const storedMessages = localStorage.getItem('ChatData');
+        if (storedMessages) {
+            setMessages(JSON.parse(storedMessages));
+        } else {
+            setMessages(messagesData.messages); // JSON에서 초기 데이터 설정
+        }
+    }, []);
+
+    // 버튼 클릭 시 사용자 ID 변경
+    const toggleUserId = () => {
+        setUserId(userId === 1 ? 2 : 1);
+    };
+
+    // 입력된 메시지 상태 (메시지 입력값 저장)
+    const [inputMessage, setInputMessage] = useState<string>("");
+
+    // 메시지 입력 시 마다 호출됨 (input 값 변화 추적)
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setInputMessage(e.target.value);
+        // 인풋의 value를 그대로 추적해서 inputMessage 상태에 담습니다!
+    };
+
+    // 스크롤을 적용을 위한 useRef. DOM 조작에는 Ref가 제일 효율적이라고 하네요!
+    const messageScrollRef = useRef<HTMLDivElement>(null);
+
+    // 메시지 전송 시 스크롤 하단 이동
+    const scrollToBottom = () => {
+        messageScrollRef.current?.scrollIntoView({ behavior: "smooth" });
+    };
+
+    // 메시지 추가 시 (전송 시) 스크롤 하단 이동 함수 실행
+    useEffect(() => {
+        scrollToBottom();
+    }, [messages]);
```

Comment:
```suggestion
useEffect(() => {
    messageScrollRef.current?.scrollIntoView({ behavior: "smooth" });
}, [messages]);

```

으로 간결하게도 가능할 거 같습니다!